### PR TITLE
macchina 0.6.0 (new formula)

### PR DESCRIPTION
**Pull request by 123marvin123**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Adds the minimalistic system fetcher "macchina"

### CODE REVIEW

File: Formula/macchina.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  desc "Basic system information fetcher, with a focus on performance and minimalism"
+  homepage "https://github.com/grtcdr/macchina"
+  url "https://github.com/grtcdr/macchina/archive/v0.6.0.tar.gz"
+  sha256 "bf6eaaca560fd2e7f04594abc6fc4615e6d3867a74816a26146f0244266cdffe"
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    Dir.chdir "macchina" do
```
Comment (alebcay) [inline Formula/macchina.rb:11]:
```suggestion
    cd "macchina" do
```

Comment (carlocab) [inline Formula/macchina.rb:11]:
```suggestion
    cd "macchina" do
```

File: Formula/macchina.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      system "cargo", "install", *std_cargo_args
+    end
+    bin.install "target/release/macchina"
```
Comment (alebcay) [inline Formula/macchina.rb:14]: Does the `cargo install` above not place the finished binary into the installation prefix?

Comment (carlocab) [inline Formula/macchina.rb:14]: Is this not already installed with `cargo install`? It is for the other formulae that use `cargo`.

File: Formula/macchina.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  test do
+    system "macchina", "--version"
```
Comment (carlocab) [inline Formula/macchina.rb:18]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — COMMENTED** [review]
Thanks for your PR, @123marvin123. I have a few comments below.

File: Formula/macchina.rb (hunk 1 above)
Comment (carlocab) [reply Formula/macchina.rb:11]: @alebcay lol.

Comment (alebcay) [reply Formula/macchina.rb:11]: I'll clean up the duplicates 😆

**Comment by carlocab** [comment]
This looks like a nice piece of software, but it currently doesn't meet the requirements for inclusion in homebrew/core. The CI log reads
```
macchina:
  * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
Error: 1 problem in 1 formula detected
```

If you wish, you can host this formula in your own tap. Here are some docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

File: Formula/macchina.rb (hunk 1 above)
Comment (123marvin123) [reply Formula/macchina.rb]: Thanks, I changed it to your suggestions :)

File: Formula/macchina.rb (hunk 2 above)
Comment (123marvin123) [reply Formula/macchina.rb]: Yes, it does :) I removed the "bin.install"

File: Formula/macchina.rb (hunk 3 above)
Comment (123marvin123) [reply Formula/macchina.rb]: > This looks like a nice piece of software, but it currently doesn't meet the requirements for inclusion in homebrew/core. The CI log reads
> 
> ```
> macchina:
>   * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
> Error: 1 problem in 1 formula detected
> ```
> 
> If you wish, you can host this formula in your own tap. Here are some docs to help you get started:
> 
> https://docs.brew.sh/Taps
> https://docs.brew.sh/Interesting-Taps-and-Forks
> https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
> https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

Oh ok, sorry I didn't know that. I'll see that I can do about hosting this in my own tap. Thank you, guys :)

**Comment by 123marvin123** [comment]
> This looks like a nice piece of software, but it currently doesn't meet the requirements for inclusion in homebrew/core. The CI log reads
> 
> ```
> macchina:
>   * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
> Error: 1 problem in 1 formula detected
> ```
> 
> If you wish, you can host this formula in your own tap. Here are some docs to help you get started:
> 
> https://docs.brew.sh/Taps
> https://docs.brew.sh/Interesting-Taps-and-Forks
> https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
> https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/


Oh ok, sorry I didn't know that. I'll see that I can do about hosting this in my own tap. Thank you, guys :)

**Comment by carlocab** [comment]
No problem. You are welcome to submit this again to homebrew/core when your software is more popular.

Thanks again for your PR!

**Pull request closed without merging**